httpgrpc/httpgrpc.go

```diff
@@ -0,0 +1,132 @@
+package httpgrpc
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
+	"github.com/mwitkow/go-grpc-middleware"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sercand/kuberesolver"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/weaveworks/common/middleware"
+)
+
+// HTTPoGRPCServer is a HTTPServer (ie gRPC) implementation
+type HTTPoGRPCServer struct {
```

Comment:
I know Go is all about the terse names, but the `o` just looks like a typo. What does it stand for? My guess is "over".